tests/unit/constraints/test_tabular.py

```diff
@@ -3189,6 +3189,31 @@ def test_is_valid_invalid(self):
         # Assert
         assert not all(result)
 
+    def test_is_valid_with_nans(self):
+        """Test the ``Range.is_valid`` when there is NaNs in the columns."""
+
+        # Setup
+        table_data_valid = pd.DataFrame({
+            'low': [1, np.nan, 3, 4, np.nan, 1],
+            'middle': [2, 3, np.nan, 5, np.nan, np.nan],
+            'high': [3, 4, 5, np.nan, 6, np.nan],
+        })
+        table_data_invalid = pd.DataFrame({
+            'low': [1, np.nan, 3, 4, np.nan, 1],
+            'middle': [2, 3, np.nan, 5, np.nan, np.nan],
+            'high': [3, 4, 2, np.nan, 6, np.nan],
+        })
+
+        instance = Range('low', 'middle', 'high')
+
+        # Run
+        result_valid = instance.is_valid(table_data_valid)
+        result_invalid = instance.is_valid(table_data_invalid)
+
+        # Assert
+        assert all(result_valid)
+        assert not all(result_invalid)
```

Comment:
Just to be safe can we directly check that the result_valid and result_invalid are the series we expect? As in compare to the expected outcome instead of jusst checking that any are false